src/main/java/mil/dds/anet/resources/EventResource.java

```diff
@@ -0,0 +1,139 @@
+    validateEvent(user, event);
+    // perform all modifications to the event and its tasks in a single transaction,
+    return executeEventUpdates(event);
+  }
+
+  /**
+   * Perform all modifications to the event and its tasks, returning the original state of the
+   * event. Should be wrapped in a single transaction to ensure consistency.
+   *
+   * @param event Event object with the desired modifications
+   * @return number of rows of the update
+   */
+  private Integer executeEventUpdates(Event event) {
+    // Verify this person has access to edit this report
+    // Either they are an author, or an approver for the current step.
+    final Event existing = dao.getByUuid(event.getUuid());
+    if (existing == null) {
+      throw new WebApplicationException("Event not found", Status.NOT_FOUND);
+    }
+    event.setDescription(Utils.isEmptyHtml(event.getDescription()) ? null
+        : Utils.sanitizeHtml(event.getDescription()));
+
+    // begin DB modifications
+    final int numRows = dao.update(event);
+    if (numRows == 0) {
+      throw new WebApplicationException("Couldn't process event update", Status.NOT_FOUND);
+    }
+
+    // Update Tasks:
+    if (event.getTasks() != null) {
+      final List<Task> existingTasks =
+          dao.getTasksForEvent(engine.getContext(), event.getUuid()).join();
+      Utils.addRemoveElementsByUuid(existingTasks, event.getTasks(),
+          newTask -> dao.addTaskToEvent(newTask, event),
+          oldTask -> dao.removeTaskFromEvent(DaoUtils.getUuid(oldTask), event));
+    }
+
+    return numRows;
+  }
+
+  private void validateEvent(final Person user, final Event event) {
+    if (event.getType() == null || event.getType().trim().isEmpty()) {
+      throw new WebApplicationException("Event type must not be empty", Status.BAD_REQUEST);
+    }
+    if (event.getName() == null || event.getName().trim().isEmpty()) {
+      throw new WebApplicationException("Event name must not be empty", Status.BAD_REQUEST);
+    }
+    if (event.getDescription() == null || event.getDescription().trim().isEmpty()) {
+      throw new WebApplicationException("Event description must not be empty", Status.BAD_REQUEST);
+    }
+    if (event.getStartDate() == null) {
+      throw new WebApplicationException("Event start date must not be empty", Status.BAD_REQUEST);
+    }
+    if (event.getEndDate() == null) {
+      throw new WebApplicationException("Event end date must not be empty", Status.BAD_REQUEST);
+    }
+    if (event.getHostOrgUuid() == null || event.getHostOrgUuid().trim().isEmpty()) {
+      throw new WebApplicationException("Event Host Organization must not be empty",
+          Status.BAD_REQUEST);
+    }
+    if (event.getAdminOrgUuid() == null || event.getAdminOrgUuid().trim().isEmpty()) {
+      throw new WebApplicationException("Event Admin Organization must not be empty",
+          Status.BAD_REQUEST);
+    }
+    assertPermission(user, event.getAdminOrgUuid());
```

Comment:
When updating you should check that the user has permission for the **original** adminOrg, as well as the one the user passed in the GraphQL mutation!